Sources/Kitura/CodableRouter+TypeSafeMiddleware.swift

```diff
@@ -0,0 +1,1245 @@
+/*
+ * Copyright IBM Corporation 2018
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Foundation
+import LoggerAPI
+import KituraNet
+import KituraContracts
+
+// Type-safe middleware Codable router
+
+extension Router {
+
+    // MARK: Codable Routing With Type-Safe Middleware
+
+    /**
+     Sets up a closure that will be invoked when a GET request to the provided route is received by the server.
+     The closure accepts a successfully executed instance of `TypeSafeMiddleware`
+     and a handler which responds with a single Codable object or a `RequestError`.
+     The handler contains the developer's logic, which determines the server's response.
+     ### Usage Example: ###
+     In this example, MySession is a struct that conforms to the TypeSafeMiddleware protocol.
+     session is an instance of MySession, that contains optional Users.
```

Comment:
Anywhere where we are starting a sentence with a variable name which is lowercase it looks poor in the jazzy API doc.

In this example, `MySession` is a struct that conforms to the `TypeSafeMiddleware` protocol and specifies an optional `User`, where `User` conforms to `Codable`.